### PR TITLE
[Intel GPU] undo broadcast on zero stride tensor for SDPA

### LINKED ISSUE #152290: [Intel GPU][PT2.8]scaled_dot_product_attention returns wrong output
### 🐛 Describe the bug

Using nightly build PT2.8, this sample code will return wrong output:
```
import torch
from datasets import load_dataset
from transformers import pipeline, Wav2Vec2Processor

model_id = "facebook/hubert-large-ls960-ft"
device = "xpu"
torch_dtype = torch.float16
generator = pipeline(
    "automatic-speech-recognition",
     model=model_id,
    device=device,
    torch_dtype=torch_dtype,
)
ds = load_dataset("patrickvonplaten/librispeech_asr_dummy", "clean", split="validation", trust_remote_code=True)
input_data = ds[0]['audio']['array']
with torch.inference_mode():
    output = generator(input_data)
print(f"output: {output}")
```
while using stock stable pytorch 2.6 version, the output is correct. After investigation, find the output of scaled_dot_product_attention API is wrong

### Versions
```
PyTorch version: 2.6.0+xpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.12 (main, Feb  4 2025, 14:57:36) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-134-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        47 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               224
On-line CPU(s) list:                  0-223
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Xeon(R) Platinum 8480+
CPU family:                           6
Model:                                143
Thread(s) per core:                   2
Core(s) per socket:                   56
Socket(s):                            2
Stepping:                             6
CPU max MHz:                          3800.0000
CPU min MHz:                          800.0000
BogoMIPS:                             4000.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust sgx bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd sgx_lc fsrm md_clear serialize tsxldtrk pconfig arch_lbr amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            5.3 MiB (112 instances)
L1i cache:                            3.5 MiB (112 instances)
L2 cache:                             224 MiB (112 instances)
L3 cache:                             210 MiB (2 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0-55,112-167
NUMA node1 CPU(s):                    56-111,168-223
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.4
[pip3] onnx==1.17.0
[pip3] pytorch-triton-xpu==3.2.0
[pip3] torch==2.6.0+xpu
[pip3] torchaudio==2.6.0+xpu
[pip3] torchvision==0.21.0+xpu
[conda] Could not collect
```

cc @gujinghui @EikanWang @fengyuan14 @guangyey

### PR DESCRIPTION
**Pull request by LuFinch**
Fix https://github.com/pytorch/pytorch/issues/152290.

The model **hubert** uses aten::expand to build attention mask by broadcasting. Pytorch uses strides[d]=0 to represent broadcast, which is not supported by oneDNN.  This PR handles this scenario.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @gujinghui @EikanWang @fengyuan14 @guangyey

### CODE REVIEW

**Review by guangyey — COMMENTED** [review]
Hey @LuFinch Please add a test case.

**Comment by LuFinch** [comment]
@guangyey I have added a UT case and already verified it.

**Comment by guangyey** [comment]
> @guangyey I have added a UT case and already verified it.

Thanks for your update!

**Review by guangyey — APPROVED** [review]

**Comment by LuFinch** [comment]
The errors `test_comprehensive__chunk_cat_xpu_bool` and `test_comprehensive__chunk_cat_xpu_float16` are not irrelevant to this PR.

**Comment by EikanWang** [comment]
@LuFinch , is there any issue with tracking the failure?

**Comment by EikanWang** [comment]
@LuFinch , meanwhile, the `sdpa` is hosted in torch. It should not be an issue of `torch-xpu-ops`. So, pls. help create an new issue in torch and then fix the issue number in this PR.

**Comment by guangyey** [comment]
Please change the description using the PyTorch issue instead.

**Comment by LuFinch** [comment]
@guangyey  Done.

**Review by guangyey — APPROVED** [review]

**Comment by guangyey** [comment]
Need @EikanWang's stamp and a final stamp from Meta.

**Comment by kaixuanliu** [comment]
Hi, @LuFinch @EikanWang @gujinghui , what is the status of this PR now? Can we accelerate the process? Thx!

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
@drisspg , may I know if you have any comments on this PR?

**Review by drisspg — APPROVED** [review]
